background.js

```diff
@@ -1,11 +1,21 @@
 chrome.browserAction.onClicked.addListener(function(tab) {
   chrome.automation.getTree(function(rootNode) {
-    if (!rootNode)
+    if (!rootNode) {
       return;
+    }
 
-    chrome.windows.create({ url: "tree.html" }, function(newWindow) {
-      var treeWindow = chrome.extension.getViews({ type: "tab", windowId: newWindow.id })[0];
-      treeWindow.createTree(rootNode);
+    chrome.tabs.onUpdated.addListener(function(tabId, processInfo, tabInfo) {
```

Comment:
How about we use the callback from chrome.windows.create to get the tabId of the newly created tab, and then use that within the callback to set up this listener if the tab is not already ready?

Also, we should remove this listener once we've successfully got a handle on `win`.